recipes/default.rb

```diff
@@ -19,19 +19,32 @@
 #
 
 return 'platform not supported' if node['platform_family'] != 'windows'
+include_recipe 'windows'
 
-include_recipe 'powershell'
+# ::Chef::Recipe.send(:include, Chef::Mixin::PowershellOut)
+::Chef::Resource::RubyBlock.send(:include, Chef::Mixin::PowershellOut)
 
-powershell 'install chocolatey' do
+powershell_script 'install chocolatey' do
   code "iex ((new-object net.webclient).DownloadString('#{node['chocolatey']['Uri']}'))"
-  not_if { ::File.exist?(::File.join(node['chocolatey']['bin_path'], 'chocolatey.bat')) }
+  convert_boolean_return true
+  not_if { ChocolateyHelpers.chocolatey_installed? }
 end
 
+ruby_block "reset ENV['ChocolateyInstall']" do
+  block do
+    cmd = powershell_out!("[System.Environment]::GetEnvironmentVariable('ChocolateyInstall', 'MACHINE')")
+    ENV['ChocolateyInstall'] = cmd.stdout.chomp
```

Comment:
@galenemery I think this allows it to gather the new environment variable, which it would not automatically update otherwise in the same run.